AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/Converters/UserBiverter.cs

```diff
@@ -0,0 +1,42 @@
+using AppSettingsManagerApi.Domain.Conversion;
+
+namespace AppSettingsManagerApi.Infrastructure.MySql.Converters;
+
+/// <summary>
+/// It's useful to have some standardized converters for objects that you'll be switching between a lot
+/// </summary>
+public class UserBiverter : IBidirectionalConverter<Model.User, User>
+{
+    private readonly IBidirectionalConverter<Model.Permission, Permission> _permissionsBiverter;
+
+    public UserBiverter(IBidirectionalConverter<Model.Permission, Permission> permissionsBiverter)
+    {
+        _permissionsBiverter = permissionsBiverter;
+    }
+
+    public User Convert(Model.User source)
+    {
+        return new User
+        {
+            UserId = source.UserId,
+            Password = source.Password,
+            Email = source.Email,
+            Permissions =
+                source.Permissions?.Select(_permissionsBiverter.Convert).ToList()
+                ?? new List<Permission>()
```

Comment:
I guess this is a matter of preference but I feel like writing source.Permissions?.Select(_permissionsBiverter.Convert).ToList()?? new List<Permission>() in one line makes it more readable. It helps me see directly what ?? is referring to. But if you feel that your method is better, then just leave it.

Reply:
And in the Permissions?.Select the ? means if Permissions is null then stop execution and just return null rather than throwing an exception. In that case then the ?? would give us the empty list on line 26